Mathlib/Data/Complex/Basic.lean

```diff
@@ -265,6 +265,18 @@ theorem I_mul_re (z : ℂ) : (I * z).re = -z.im := by simp
 
 theorem I_mul_im (z : ℂ) : (I * z).im = z.re := by simp
 
+theorem mul_I_re_ofReal (z : ℝ) : (z * I).re = 0 := by
+  rw [mul_I_re, ofReal_im, neg_zero]
```

Comment:
All of these are currently provable using `by simp`. Do we really need them?

Reply:
I moved them into #23165 and then decided against them.